### ISSUE TITLE
Feature Request: Full sslPolicy support for azurerm_application_gateway

### ISSUE THREAD

**Issue by cdhunt** [terraform-providers/terraform-provider-azurerm#1536]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

It looks like a single property (`disabledSslProtocols`) of the sslPolicy configuration object was added to the resource. We would like access to the full object.

### New or Affected Resource(s)

* azurerm_application_gateway

### Potential Terraform Configuration

```hcl
  ssl_policy {
    disabled_ssl_protocols = list
    policy_type = string
    policy_name = string
    cipher_suites = list
    min_protocol_version = list
  }
```

### References

* https://docs.microsoft.com/en-us/azure/templates/microsoft.network/applicationgateways#applicationgatewaysslpolicy-object

**Comment by stack72** [comment]
@katbyte / @tombuildsstuff is this a duplicate of this? https://github.com/terraform-providers/terraform-provider-azurerm/issues/619

**Comment by tombuildsstuff** [comment]
Duplicate of #619 - thanks for pointing that out @stack72 :)

**Comment by cdhunt** [comment]
There's no mention of sslPolicy in #619. Will that be included in that work?

**Comment by tombuildsstuff** [comment]
Reopening this to split this out from #1576

**Comment by cdhunt** [comment]
Are there still blockers for this or is it now technically possible?

**Comment by tombuildsstuff** [comment]
@cdhunt I believe it should be possible for someone to implement this once #2054 is merged - it requires a little thought around deprecating the existing `disabled_ssl_protocols` block (as we'll want to move it into the new `ssl_policy` block) probably as a part of 2.0

**Comment by cdhunt** [comment]
Yeah, dealing with the existing `disabled_ssl_protocols` block without a breaking change was my concern. Thanks.

**Comment by bs-matil** [comment]
I will soon file a PR regards this topic. I feel our Implementation which was done a while ago before this discussion may not fully fit. But I will adopt it in any way needed to get it upstream. See ya later :).

**Comment by jstewart612** [comment]
@bs-matil has this work begun yet?  I am curious because, if there is partial work complete, I was going to expand from that and submit a PR.  If not, I am considering taking a crack at this myself from scratch:  my organization has a large number of AAGs and it's a real PITA not having this.

**Comment by GCole64** [comment]
👍

**Comment by GCole64** [comment]
If you want by all means

**Comment by bs-matil** [comment]
@jstewart612 I am about to publish the work today

**Comment by bs-matil** [comment]
@tombuildsstuff I submitted a PR under #3360
I expect changes are needed I'll wait for input :)
